src/transport_ofi.h

```diff
@@ -488,6 +488,8 @@ int shmem_transport_fence(shmem_transport_ctx_t* ctx)
      * non-fetching atomics to be completed in order to ensure ordering. */
     shmem_transport_put_quiet(ctx);
 #endif
+    /* Complete fetching ops; needed to support nonblocking fetch-atomics */
+    shmem_transport_get_wait(ctx);
 
```

Comment:
@jdinan Was this a bug earlier? This seems to be covering any fetch operations now.

Reply:
This was a deliberate optimization, since fence did not need to complete anything on the get counter.  With the addition of fetch-atomics, this is required to order the atomic update.  This change does have the unfortunate side effect of making fence slower in instances where nonblocking get operations are now completed, but were not previously.